x-pack/plugin/esql/compute/test/src/main/java/org/elasticsearch/compute/test/OperatorTestCase.java

```diff
@@ -77,6 +79,19 @@ public final void testSimpleLargeInput() {
         assertSimple(driverContext(), between(1_000, 10_000));
     }
 
+    /**
+     * Test with a local breaker
+     */
+    public final void testWithLocalBreaker() {
+        BlockFactory blockFactory = blockFactory();
+        final int overReservedBytes = between(0, 1024 * 1024);
+        final int maxOverReservedBytes = between(overReservedBytes, 1024 * 1024);
+        var localBreaker = new LocalCircuitBreaker(blockFactory.breaker(), overReservedBytes, maxOverReservedBytes);
```

Comment:
Should we wrap this in something that `assert`s that we're on the same thread every time?

Reply:
++ I pushed https://github.com/elastic/elasticsearch/pull/128162/commits/ce545830587ed9623878ec2c502ab6cffa7425e7.